Tests/Marketplace/marketplace_services.py

```diff
@@ -2126,217 +2179,266 @@ def collect_content_items(self):
                             'marketplaces': content_item.get('marketplaces', ["xsoar", "marketplacev2"]),
-                        })
+                        }
 
                     elif current_directory == PackFolders.XDRC_TEMPLATES.value and pack_file_name.startswith("external-"):
                         self.add_pack_type_tags(content_item, 'XDRCTemplate')
-                        folder_collected_items.append({
+                        metadata_output = {
                             'id': content_item.get('content_global_id', ''),
                             'content_global_id': content_item.get('content_global_id', ''),
                             'name': content_item.get('name', ''),
                             'os_type': content_item.get('os_type', ''),
                             'profile_type': content_item.get('profile_type', ''),
                             'marketplaces': content_item.get('marketplaces', ["marketplacev2"]),
-                        })
+                            'fromversion': latest_fromversion,
+                            'toversion': latest_toversion,
+                        }
 
                     elif current_directory == PackFolders.LAYOUT_RULES.value and pack_file_name.startswith(
                             "external-"):
                         self.add_pack_type_tags(content_item, 'LayoutRule')
-                        layout_rule_metadata = {
+                        metadata_output = {
                             'id': content_item.get('rule_id', ''),
                             'name': content_item.get('rule_name', ''),
                             'layout_id': content_item.get('layout_id', ''),
                             'marketplaces': content_item.get('marketplaces', ["marketplacev2"]),
+                            'fromversion': latest_fromversion,
+                            'toversion': latest_toversion,
                         }
                         layout_rule_description = content_item.get('description')
                         if layout_rule_description is not None:
-                            layout_rule_metadata['description'] = layout_rule_description
-                        folder_collected_items.append(layout_rule_metadata)
-
+                            metadata_output['description'] = layout_rule_description
                     else:
                         logging.info(f'Failed to collect: {current_directory}')
+                        continue
+                    if replace_content_item:
+                        folder_collected_items = [metadata_output
+                                                  if d["id"] == metadata_output["id"]
+                                                  else
+                                                  d for d in folder_collected_items]
+                    if latest_toversion == '':
+                        continue
+                    else:
+                        folder_collected_items.append(metadata_output)
```

Comment:
Why are we adding the metadata only if `latest_toversion != ''`?

Reply:
latest_toversion refers to the latest toversion saved to the map and list under that content item id.
if we only got the empty toversion it means we got the latest content item / the only content item and there is no need to add another metadata for that content item.